ostia-operator/test/integration.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+
+set -m
+EXIT_CODE=0
+
+test_perms () {
+    [[ "$(oc auth can-i ${1} ${2})" == "yes" ]] && return
+    echo "Current user cannot ${1} ${2}"
+    exit 1
+}
+
+checkexit() {
+   if [[ $? > 0 ]]; then
+      echo "at least one test failed";
+      EXIT_CODE=1;
+   fi
+}
+
+if $(oc get pods > /dev/null 2>&1); then
+    HOSTNAME_FROM_OC=$(basename $(oc whoami --show-server=true) | cut -f1 -d":")
+    [[ ${HOSTNAME_FROM_OC} == *.nip.io ]] || HOSTNAME_FROM_OC+=.nip.io
+fi
+
+if [ -z "$OPENSHIFT_PUBLIC_HOSTNAME" ]; then
+    export OPENSHIFT_PUBLIC_HOSTNAME=${HOSTNAME_FROM_OC}
+fi
+
+
+dir=$(pwd)
+
+test_perms create CustomResourceDefinition
+oc create -f ../deploy/crd.yaml &> /dev/null || true
+test_perms create RoleBinding
+
+trap 'checkexit $? $$' CHLD
+
+for f in ./tests/*-*/*.sh; do
+  echo "Launching: $f"
+  cd ${f%/*}; ./$(basename $f) &
+  cd $dir
+done
```

Comment:
here there is an opportunity to use bash's associative arrays to map pids to specific `$f`s and point out exactly which tests failed by pushing each failed exit status' $f job to another array.

Reply:
Tried using associative array on osx got the following error: 

```
./integration.sh: line 4: declare: -A: invalid option
declare: usage: declare [-afFirtx] [-p] [name[=value] ...]
```

```
bash --version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin17)
```
Believe we need version 4 which doesn't seem to be installed by default

Reply:
Also wondering do you have any suggestions on the best way to capture the exit code of the subprocess' being run in parallel?

Reply:
I'm definitely against bash specific functions. Almost all the scripts we have can be just basic shell using /bin/sh.

Reply:
We should work on the output from the scripts, store the output in a debug.log and only show what failed in stdout ... wdyt?

Adding an "output" function that prefixes everything with the current folder name (test name) would be nice because there are multiple scripts at the same time.. is that, or adding the test name to each "echo"

Reply:
@PhilipGough when you launch the backgrounded test its pid will be stored in `$!`, so you can just add it to some collection. Wrt associative arrays they were added in version 4 of bash - requiring a lower version means you will need to write a bit more code, for example by assigning positions like a structure of arrays: pids[4] will match with jobs[4] and exitvalues[4]. Avoiding bashisms (ie. using posix shell scripting alone) complicates life a whole lot more (ie. you don't even have arrays).